src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,19 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
+if PY3 and pathlib is not None:  # pragma: no branch
+    _global_type_lookup[pathlib.Path] = \
```

Comment:
Sure - try `from_type(pathlib.PosixPath).example()`.  This will only work if a strategy is registered for PosixPath.  Registering something *only* for the child type also works:

```
path_type = pathlib.WindowsPath if os.name == 'nt' else pathlib.PosixPath
_global_type_lookup[path_type] = \
    st.fspaths(allow_pathlike=False).map(os.fsdecode).map(path_type)
```

This will resolve Path and PurePath on all platforms, and the os-specific variants on the relevant operating system.